src/glossary.md

```diff
@@ -11,18 +11,19 @@ completeness            |  completeness is a technical term in type theory. Comp
 DefId                   |  an index identifying a definition (see `librustc/hir/def_id.rs`). Uniquely identifies a `DefPath`.
-HIR                     |  the High-level IR, created by lowering and desugaring the AST ([see more](hir.html))
-HirId                   |  identifies a particular node in the HIR by combining a def-id with an "intra-definition offset".
 'gcx                    |  the lifetime of the global arena ([see more](ty.html))
 generics                |  the set of generic type parameters defined on a type or item
+HIR                     |  the High-level IR, created by lowering and desugaring the AST ([see more](hir.html))
+HirId                   |  identifies a particular node in the HIR by combining a def-id with an "intra-definition offset".
 ICE                     |  internal compiler error. When the compiler crashes.
 ICH                     |  incremental compilation hash. ICHs are used as fingerprints for things such as HIR and crate metadata, to check if changes have been made. This is useful in incremental compilation to see if part of a crate has changed and should be recompiled.
 infcx                   |  the inference context (see `librustc/infer`)
-MIR                     |  the Mid-level IR that is created after type-checking for use by borrowck and trans ([see more](./mir.html))
-miri                    |  an interpreter for MIR used for constant evaluation ([see more](./miri.html))
-obligation              |  something that must be proven by the trait system ([see more](trait-resolution.html))
+IR                      |  Intermediate Representation. A general term in compilers. During compilation, the code is transformed from raw source (ASCII text) to various IRs. In Rust, these are primarily HIR, MIR, and LLVM IR. Each IR is well-suited for some set of computations. For example, MIR is well-suited for the borrow checker, and LLVM IR is well-suited for codegen because LLVM accepts it.
 local crate             |  the crate currently being compiled.
+LTO                     |  Link-Time Optimizations. A set of optimizations offered by LLVM that occur just before the final binary is linked. These include optmizations like removing functions that are never used in the final program, for example. _ThinLTO_ is a variant of LTO that aims to be a bit more scalable and efficient, but possibly sacrifices some optimizations. You may also read issues in the Rust repo about "FatLTO", which is the loving nickname given to non-Thin LTO. LLVM documentation: [here][lto] and [here][thinlto]
+[LLVM]                  |  (actually not an acronym :P) an open-source compiler backend. It accepts LLVM IR and outputs native binaries. Various languages (e.g. Rust) can then implement a compiler front-end that output LLVM IR and use LLVM to compile to all the platforms LLVM supports.
```

Comment:
Doesn't it stand for Low Level Virtual Machine? Or did they do it away with that.

Reply:
That's what I thought too, "The name "LLVM" itself is not an acronym; it is the full name of the project."

Reply:
I sort of remember that it was an acronym and they decided that was dumb, since it wasn't really a VM anymore, and so they stopped that.

Reply:
Ah, I can change it if you want. I just wrote whatever they said on the website...

Reply:
no it's fine :)